src/stan/lang/function_signatures.h

```diff
@@ -123,24 +123,43 @@ for (size_t i = 0; i < vector_types.size(); ++i) {
     }
   }
 }
-add_binary("beta_rng");
+for (size_t i = 0; i < primitive_types.size(); ++i) {
+  for (size_t j = 0; j < primitive_types.size(); ++j) {
+    add("beta_rng", expr_type(double_type()), primitive_types[i], primitive_types[j]);
+  }
+ }
+for (size_t i = 0; i < primitive_types.size(); i++) {
+  add("beta_rng", expr_type(vector_type()), primitive_types[i], expr_type(int_type(), 1U));
```

Comment:
Are `int[]` not allowed with other vector arguments?

Reply:
Please correct, if wrong.  My going assumption was that `expr_type(int_type(), 1U)` meant int[].  In which case, lines 140 and 141 have int[] with other vector arguments.

Reply:
Yes, that's right, `expr_type(int_type(), 1U)` gives you int arrays.  The whole type system's a bit more opaque than it should be and not well abstracted for specifying types for these kinds of polymorphic functions.   Sorry it's such a pain to deal with.